server/src/objects/fakehub.rs

```diff
@@ -0,0 +1,143 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2024 Aliaksei Bialiauski
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included
+// in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+use crate::objects::github::GitHub;
+use crate::objects::user::User;
+use std::collections::HashMap;
+use uuid::Uuid;
+
+/// Fakehub.
+pub struct Fakehub {
+    hubs: HashMap<Uuid, GitHub>,
+    urled: HashMap<String, GitHub>,
+}
+
+impl Default for Fakehub {
+    fn default() -> Fakehub {
+        let mut hubs: HashMap<Uuid, GitHub> = HashMap::new();
+        let id = Uuid::new_v4();
+        hubs.insert(
+            id,
+            GitHub {
+                id,
+                url: String::from("https://github.com"),
+                users: vec![User::new(String::from("jeff"))],
+            },
+        );
+        Fakehub {
+            hubs: hubs.clone(),
+            urled: hubs.into_values().map(|v| (v.clone().url, v)).collect(),
+        }
+    }
+}
+
+impl Fakehub {
+    /// Add new GitHub
+    pub fn add(&mut self, github: GitHub) {
+        self.hubs.insert(github.id, github.clone());
+        self.urled.insert(github.clone().url, github);
+    }
+
+    /// All GitHub instances.
+    pub fn githubs(self) -> Vec<GitHub> {
+        self.hubs
+            .iter()
+            .map(|(k, v)| GitHub {
+                id: *k,
+                url: v.url.clone(),
+                users: v.users.clone(),
+            })
+            .collect()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::objects::fakehub::Fakehub;
+    use crate::objects::github::GitHub;
+    use anyhow::Result;
+    use hamcrest::{equal_to, is, HamcrestMatcher};
+    use uuid::Uuid;
+
+    #[test]
+    fn creates_default_fakehub() -> Result<()> {
+        let fakehub = Fakehub::default();
+        let default = fakehub
+            .urled
+            .get("https://github.com")
```

Comment:
to run these tests internet connection is required. I think it's not the best approach to use in testing tool _that provides a functionality of github without connection to real github_

Reply:
@l3r8yJ actually we just accessing HashMap called `urled` in order to get GitHub object by given URL. Got the idea?

Reply:
@h1alexbel sure, but why do we need an url then?

Reply:
@l3r8yJ I think it is both: 1) unique attribute, so we can distinguish GitHubs between each other, and 2) convenient to use, so we don't need to remember ID, or other information, just URL.